src/Components/Components/src/Rendering/RenderTreeBuilder.cs

```diff
@@ -175,6 +175,11 @@ public void AddAttribute(int sequence, string name)
             throw new InvalidOperationException($"Valueless attributes may only be added immediately after frames of type {RenderTreeFrameType.Element}");
         }
 
+        if (TrackNamedEventHandlers && string.Equals(name, "@onsubmit:name", StringComparison.Ordinal))
+        {
+            _entries.AppendAttribute(sequence, name, "");
```

Comment:
Is the idea that people would write `<... @onsubmit:name>` with no value?

Reply:
You can do no value, but more commonly you would do `@onsubmit:name=""` which gets translated to `true`, which is what this code avoids. Once the directive for the razor compiler is in, this will just be `SetEventHandlerName`